File: registratorapp/src/main/java/de/weimarnetz/registrator/configuration/SecurityConfig.java
package de.weimarnetz.registrator.configuration;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.http.HttpMethod;
import org.springframework.security.config.annotation.web.builders.HttpSecurity;
import org.springframework.security.config.annotation.web.builders.WebSecurity;
import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
import org.springframework.security.config.http.SessionCreationPolicy;
import org.springframework.security.core.userdetails.User;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.security.provisioning.InMemoryUserDetailsManager;

@Configuration
public class SecurityConfig extends WebSecurityConfigurerAdapter {
    static final String ADMIN_ROLE = "ADMIN";

    @Value("${delete.user}")
    private String username;
    @Value("${delete.password}")
    private String password;

    @Override
    public void configure(WebSecurity web) {
        web
                .ignoring()
                .regexMatchers("/(css|js|fonts)/.*")
                .antMatchers(HttpMethod.GET)
                .antMatchers(HttpMethod.POST)
                .antMatchers(HttpMethod.PUT)
                .antMatchers(HttpMethod.OPTIONS);
    }

    @Override
    protected void configure(HttpSecurity http) throws Exception {
        http
                .headers().frameOptions().sameOrigin()
                .and()
                .sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS)
                .and()
                .httpBasic()
                .and()
                .csrf().disable()
                .authorizeRequests()
                .antMatchers(HttpMethod.DELETE).hasRole(ADMIN_ROLE);
    }


    @Bean
    public UserDetailsService userDetailsService() {
        // Set the inMemoryAuthentication object with the given credentials:
        InMemoryUserDetailsManager manager = new InMemoryUserDetailsManager();
        String encodedPassword = passwordEncoder().encode(password);
        manager.createUser(User.withUsername(username).password(encodedPassword).roles(ADMIN_ROLE).build());
        return manager;
    }

    @Bean
    public PasswordEncoder passwordEncoder() {
        return new BCryptPasswordEncoder();
    }
}

File: src/main/java/de/weimarnetz/registrator/configuration/SecurityConfig.kt
package de.weimarnetz.registrator.configuration

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.http.HttpMethod
import org.springframework.security.config.annotation.web.builders.HttpSecurity
import org.springframework.security.config.http.SessionCreationPolicy
import org.springframework.security.core.userdetails.User
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.security.provisioning.InMemoryUserDetailsManager
import org.springframework.security.web.SecurityFilterChain


const val ADMIN_ROLE = "ADMIN"

@Configuration
class SecurityConfig {
    @Value("\${admin.user}")
    private val username: String? = null

    @Value("\${admin.password}")
    private val password: String? = null

    @Bean
    fun configure(http: HttpSecurity): SecurityFilterChain {
        return http
            .headers { headersConfigurer ->
                headersConfigurer.frameOptions { it.sameOrigin() }
            }.sessionManagement {
                it.sessionCreationPolicy(SessionCreationPolicy.STATELESS)
            }.csrf { it.disable() }
            .httpBasic {}
            .authorizeHttpRequests {
                it.requestMatchers(HttpMethod.GET, "/dumpDatabase").hasRole(ADMIN_ROLE)
                it.requestMatchers(HttpMethod.GET).permitAll()
                it.requestMatchers(HttpMethod.POST, "/importDatabase").hasRole(ADMIN_ROLE)
                it.requestMatchers(HttpMethod.POST).permitAll()
                it.requestMatchers(HttpMethod.PUT).permitAll()
                it.requestMatchers(HttpMethod.OPTIONS).permitAll()
                it.requestMatchers("/(css|js|fonts)/.*").permitAll()
                it.requestMatchers(HttpMethod.DELETE).hasRole(ADMIN_ROLE)
                it.anyRequest().authenticated()
            }.build()

    }


    @Bean
    fun userDetailsService(): InMemoryUserDetailsManager {
        val encodedPassword = passwordEncoder().encode(password)
        val user = User
            .withUsername(username)
            .password(encodedPassword)
            .roles(ADMIN_ROLE)
            .build()
        return InMemoryUserDetailsManager(user)
    }

    @Bean
    fun passwordEncoder(): PasswordEncoder {
        return BCryptPasswordEncoder()
    }
}